pennylane_orquestra/gen_workflow.py

```diff
@@ -0,0 +1,172 @@
+}
+
+qulacs_import = {
+    "name": "qe-qulacs",
+    "type": "git",
+    "parameters": {"repository": "git@github.com:zapatacomputing/qe-qulacs.git", "branch": "dev"},
+}
+
+backend_import_db = {
+    "qe-forest": forest_import,
+    "qe-qiskit": qiskit_import,
+    "qe-qulacs": qulacs_import,
+}
+
+
+def step_dictionary(name_suffix):
+    """Creates a new step with a pre-defined name suffixed with the name
+    passed.
+
+    Args:
+        name_suffix (str): The name suffix to use, usually the index of the
+            step. Such an index as suffix can be used for sorting the workflow
+            steps (e.g., for batch execution for the Orquestra device).
+
+    Returns:
+        dict: the dictionary containing information for the step
+    """
+    name = "run-circuit-and-get-expval-" + name_suffix
+    step_dict = {
+        "name": name,
+        "config": {
+            "runtime": {
+                "language": "python3",
+                "imports": [
+                    "pl_orquestra",
+                    "z-quantum-core",
+                    "qe-openfermion",
+                    # Place to insert: step backend component import
+                ],
+                "parameters": {
+                    "file": "pl_orquestra/steps/expval.py",
+                    "function": "run_circuit_and_get_expval",
+                },
+            }
+        },
+        # Place to insert: inputs
+        "outputs": [{"name": "expval", "type": "expval", "path": "/app/expval.json"}],
+    }
+
+    return step_dict
+
+
+def gen_expval_workflow(component, backend_specs, circuits, operators, **kwargs):
+    """Workflow template for computing the expectation value of operators
+    given a quantum circuit and a device backend.
+
+    Args:
+        component (str): the name of the Orquestra component to use
+        backend_specs (str): the Orquestra backend specifications as a json
+            string
+        circuits (list): list of OpenQASM 2.0 programs, each representing a
+            circuit as an input for a workflow step
+        operators (list): A list of json strings, each representing a list of
+            operators as an input for a workflow step. Each operator is a string in
+            an ``openfermion.QubitOperator`` or ``openfermion.IsingOperator``
+            representation. For example, ``['["1 [Z0]", "1 [Z1]"]']`` is an
+            input for a single step that returns the expectation value of two
+            observables: ``Z0`` and ``Z1``.
+
+    Keyword arguments:
+        resources=None (str): the machine resources to use for executing the
+            workflow
+
+    Returns:
+        dict: the dictionary that contains the workflow template to be
+        submitted to Orquestra
+    """
+    backend_import = backend_import_db.get(component, None)
+    if backend_import is None:
+        raise ValueError("The specified backend component is not supported.")
+
+    expval_template = {
```

Comment:
Out of curiosity, why not make this a module variable?

Reply:
There are mutable entries inside of the dictionary, thought its safer this way.